llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -6003,6 +6003,13 @@ void computeKnownFPClass(const Value *V, const APInt &DemandedElts,
         if (IncValue == P)
           continue;
 
+        // If the Use is a select of this phi, use the fp class of the other
+        // operand to break the recursion.
+        Value *V;
+        if (match(IncValue, m_Select(m_Value(), m_Specific(P), m_Value(V))) ||
+            match(IncValue, m_Select(m_Value(), m_Value(V), m_Specific(P))))
```

Comment:
Just a nit, but maybe this means we should add `m_c_Select` matcher.

Reply:
It would be hard to represent that with the negated condition I think

Reply:
Agreed, think it's probably not practical